ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js

```diff
@@ -50,34 +70,101 @@ export const RepositoryForm = ({ title }) => {
     }
   }, [dispatch, repositoriesLoaded]);
 
+  const typeString = type === "ppa" ? "PPA" : "repository";
+  let initialValues;
+  let title;
+  if (repository) {
+    title = `Edit ${typeString}`;
+    initialValues = {
+      arches: repository.arches,
+      components: repository.components.join(", "),
+      default: repository.default,
+      disable_sources: repository.disable_sources,
+      disabled_components: repository.disabled_components,
+      disabled_pockets: repository.disabled_pockets,
+      distributions: repository.distributions.join(", "),
+      enabled: repository.enabled,
+      key: repository.key,
+      name: repository.name,
+      url: repository.url
+    };
+  } else {
+    title = `Add ${typeString}`;
+    initialValues = {
+      arches: ["i386", "amd64"],
+      components: "",
+      default: false,
+      disable_sources: false,
+      disabled_components: [],
+      disabled_pockets: [],
+      distributions: "",
+      enabled: true,
+      key: "",
+      name: "",
+      url: type === "ppa" ? "ppa:" : ""
+    };
+  }
+
   return (
     <>
       {!allLoaded ? (
         <Loader text="Loading..." />
       ) : (
         <Card highlighted className="repo-form">
           <Row>
-            <Col size="3">
+            <Col size="2">
               <h4>{title}</h4>
             </Col>
-            <Col size="7">
-              <div className="repo-form__buttons">
-                <Button
-                  appearance="base"
-                  className="u-no-margin--bottom"
-                  onClick={() => window.history.back()}
-                  type="button"
-                >
-                  Cancel
-                </Button>
-                <Button
-                  appearance="positive"
-                  className="u-no-margin--bottom"
-                  type="button"
-                >
-                  Save repository
-                </Button>
-              </div>
+            <Col size="8">
+              <Formik
+                initialValues={initialValues}
+                validationSchema={RepositorySchema}
+                onSubmit={(values, { resetForm }) => {
+                  const payload = {
+                    arches: values.arches,
+                    components: values.components.split(" ,").filter(Boolean),
+                    default: values.default,
+                    disable_sources: values.disable_sources,
+                    disabled_components: values.disabled_components,
+                    disabled_pockets: values.disabled_pockets,
+                    distributions: values.distributions
+                      .split(" ,")
+                      .filter(Boolean),
+                    enabled: values.enabled,
+                    key: values.key,
+                    name: values.name,
+                    url: values.url
+                  };
+                  console.log(payload);
```

Comment:
Stray `console.log` here

Reply:
@squidsoup Replaced with an alert :P. It's just there temporarily until #160 is implemented